### ISSUE TITLE
Modify use of time_sync() for obtaining time

### ISSUE THREAD

**Issue by jrmses** [ultralytics/yolov5#6600]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar feature requests.

### Description

In the code you are using time_sync() for getting time. For example in line 116 of detect.py.
time_sync() creates a forced syncronization if you are using Cuda:

def time_sync():
    # pytorch-accurate time
    if torch.cuda.is_available():
        torch.cuda.synchronize()
    return time.time()

The use of this produces a forced syncronization in Cuda that damages overall performance.
Acording to pytorch documentation [https://pytorch.org/docs/stable/generated/torch.cuda.synchronize.html] : "Waits for all kernels in all streams on a CUDA device to complete." You can measure effect easily if for run the code.

Sugestion is to remove it (by using time1=time.time() for example)

### Use case

Improve overall performance

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@jrmses unfortunately CUDA profiling will be incorrect without syncing. Do you have any before and after profiling results that show improvement from your recommended change?

**Comment by jrmses** [comment]
Hi again. No, I dont have with me now, sorry for that. I made some test with my developments that included Yolo5 and it was noticiable, but I dont have exact numbers with me now.

**Comment by glenn-jocher** [comment]
@jrmses see https://discuss.pytorch.org/t/why-time-time-in-python-is-inaccurte/94274

**Comment by jrmses** [comment]
Yes, I was aware of that. This is the reason I put that as "enhacenment". On the other hand, at least for me, pure performance is more important than accuracy in timing. As a secundary suggestion maybe it could make sense to put it as "if __debug__:" or similar (there are two "__" before debug, but the editor converts it to bold text)

**Comment by glenn-jocher** [comment]
@jrmses I don't see any benefit in my tests. If you can demonstrate otherwise we'll consider it of course.

```python
# PyTorch Hub
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')
model_test = torch.hub.load('ultralytics/yolov5:test/time_sync', 'yolov5s')

# Images
dir = 'data/images/'
imgs = [dir + f for f in ('zidane.jpg', 'bus.jpg')]  # batch of images

# Inference
for _ in range(1):
  results = model(imgs)
  %timeit model(imgs)
  results = model_test(imgs)
  %timeit model_test(imgs)
```

### Colab results
<img width="766" alt="Screenshot 2022-02-10 at 15 38 11" src="https://user-images.githubusercontent.com/26833433/153430113-5b7e3705-6596-44e5-ab69-7af66cc4e4db.png">

**Comment by glenn-jocher** [comment]
https://github.com/ultralytics/yolov5/compare/master...test/time_sync

**Comment by jrmses** [comment]
Unfortunately I dont have numbers. In my case I used it when processing video footage in "real" time and them using Deepsort for tracking. (reading live videos from YouTube). I will try to provide some numbers. Thank you for your time in any case.
